File: src/main/kotlin/algorithms/easy/TwoSum.kt
package br.com.rbthmn.algorithms.easy

import br.com.rbthmn.Problem

@Problem(1)
fun twoSum(nums: IntArray, target: Int): IntArray {
    var i = 0
    while (i < nums.size) {
        var j = nums.size - 1
        while (j > i) {
            if (nums[i] + nums[j] == target) return intArrayOf(i, j)
            else j--
        }
        i++
    }
    return intArrayOf()
}

File: src/main/kotlin/algorithms/easy/TwoSum.js
var twoSum = function (nums, target) {
    for (let i = 0; i < nums.length - 1; i++) {
        let counter = nums[i]
        let indexes = [i]
        for (let y = i + 1; y < nums.length; y++) {
            counter += nums[y]
            indexes.push(y)

            if (counter == target) {
                return indexes
            } else {
                counter -= nums[y]
                indexes.pop()
            }
        }
    }
};